### PR TITLE
Added idl2library cmake helper.

### PR DESCRIPTION
**Pull request by doug1234**

### CODE REVIEW

File: idl2library.cmake [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+    list(REMOVE_DUPLICATES ALL_IDL_RELDIRS)
+    list(REMOVE_DUPLICATES ALL_IDL_PATHS)
+
+    #Add a library to build for each input idl
+    foreach(SINGLE_IDL ${IDL_WISHLIST})
+        get_idl_target_name(${SINGLE_IDL} current_idl_target)
+
+        # If this target has already been created, we're done
+        list(FIND IDL_TARGETS ${current_idl_target} SEARCH_INDEX)
+        if(${SEARCH_INDEX} GREATER -1)
```
Comment (iguessthislldo) [inline idl2library.cmake]: Targets can be tested directly using [`if(TARGET <target-name>)`](https://cmake.org/cmake/help/latest/command/if.html#target)

File: idl2library.cmake [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    get_filename_component(current_idl_shortname ${idlfile} NAME_WE)
+    string(TOLOWER "${current_idl_shortname}" current_idl_shortname)
+    set(temp_output "${current_idl_shortname}")
+    string(APPEND temp_output "${IDL_TARGET_SUFFIX}")
+    set(${output} ${temp_output})
+endmacro()
+
+#Finds all of the input file's (absolute path) dependencies and returns them in IDL_TARGET_DEPENDENCIES
+function(find_idl_dependencies input_file)	
+	unset(search_list)
```
Comment (iguessthislldo) [inline idl2library.cmake]: This file is mixing spaces and tabs for indents.

File: idl2library.cmake [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+        if(IDL_ARG_EXTENSION STREQUAL "")
+            #If they didn't then add it..
+            set(IDL_ARG "${IDL_ARG}.idl")
+        endif()
+        list(APPEND IDL_WISHLIST ${IDL_ARG})
+    endforeach()
+    
+    option(OPENDDS_CPP11_IDL_MAPPING "Use C++11 IDL mapping" OFF)	
+    option(OPENDDS_CMAKE_VERBOSE "Print verbose output when loading the OpenDDS Config Package" ON)
+    option(OPENDDS_ALLOW_ENV_CHANGE "Projects may reset OpenDDS env vars." ON)
```
Comment (iguessthislldo) [inline idl2library.cmake]:
```suggestion
```

This hasn't done anything for a long time.

File: idl2library.cmake [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+			TAO_IDL_OPTIONS ${current_idl_include_opts}
+        )
+        target_link_libraries(${current_idl_target}
+           ${IDL_TARGET_DEPENDENCIES}
+            OpenDDS::Dcps
+        )
+
+        # Group the IDL projects together
+        set_target_properties(${current_idl_target} PROPERTIES FOLDER IDL)
+        target_compile_definitions(${current_idl_target} PUBLIC __OPENDDS_IDL_HAS_ANNOTATIONS _HAS_AUTO_PTR_ETC=1 _SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING)
```
Comment (iguessthislldo) [inline idl2library.cmake]:
```suggestion
        target_compile_definitions(${current_idl_target} PUBLIC  _HAS_AUTO_PTR_ETC=1 _SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING)
```

I'm not sure what this is supposed to do. `__OPENDDS_IDL_HAS_ANNOTATIONS` is for IDL and this function is for C++ macro definitions.

File: idl2library.cmake [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+	unset(search_list)
+	unset(return_list)
+	LIST(APPEND current_list ${input_file}) #Files that need to be searched (full path)
+	LIST(APPEND search_list ${input_file}) #Don't search more than once (full path)
+	
+	while(current_list)
+	    SET(file_list ${current_list})
+		unset(current_list)
+		foreach(current_file ${file_list})
+			set (INCLUDE_REGEX "^[ \t]*\#include[ \t]+\"(.+\\.[I|i][D|d][L|l])\".*$")
```
Comment (iguessthislldo) [inline idl2library.cmake]: Preprocessor statements can have spaces after the # and includes can use `<>` as well as `""`.  Even then trying to parse anything in an IDL file outside an IDL compiler doesn't seem very robust.

**Review by iguessthislldo — CHANGES_REQUESTED** [review]

File: idl2library.cmake (hunk 1 above)
Comment (doug1234) [reply idl2library.cmake]: Made the change.

File: idl2library.cmake (hunk 2 above)
Comment (doug1234) [reply idl2library.cmake]: Made the file consistent.

File: idl2library.cmake (hunk 3 above)
Comment (doug1234) [reply idl2library.cmake]: Removed.

File: idl2library.cmake (hunk 4 above)
Comment (doug1234) [reply idl2library.cmake]: Removed.

File: idl2library.cmake (hunk 5 above)
Comment (doug1234) [reply idl2library.cmake]: Made the two suggested tweaks.  I agree about robustness but it is useful enough to keep and has worked for a while.

**Review by iguessthislldo — APPROVED** [review]

**Pull request merged**